### PR TITLE
Update 'rabbitmq-c' to v0.10.0-773b883

### LINKED ISSUE #345: RabbitMQ package issue
* I've read [Brief overview section](https://hunter.readthedocs.io/en/latest/overview.html) and do understand basic concepts. **[Yes|No]**
* I've read [F.A.Q. section](https://hunter.readthedocs.io/en/latest/faq.html) and there is no solution to my problem there. **[Yes|No]**
* I've read [Code of Conduct](https://github.com/cpp-pm/hunter/blob/master/.github/CODE_OF_CONDUCT.md), I promise to be polite and will do my best at being constructive. **[Yes|No]**
* I've read [Reporting bugs section](https://hunter.readthedocs.io/en/latest/contributing.html#reporting-bugs) carefully. **[Yes|No]**
* I've checked that all the `hunter_add_package`/`find_package` API used by me in the example is the same as in [documentation](https://hunter.readthedocs.io/en/latest/packages.html). **[Yes|No]**
* I'm using [latest](https://github.com/cpp-pm/hunter/releases) Hunter `URL`/`SHA1`. **[Yes|No]**

While trying to use the package `rabbitmq-c`, CMake is unable to find and link the library. However, Hunter is able to download and build it properly. When testing the example within the Hunter repository I get the following output:

```
❯ cmake -DCMAKE_BUILD_TYPE=Debug ..
Including HunterGate: /Users/sgerbino/Projects/hunter/gate/cmake/HunterGate.cmake
-- The C compiler identification is AppleClang 12.0.0.12000032
-- The CXX compiler identification is AppleClang 12.0.0.12000032
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- [hunter] Calculating Toolchain-SHA1
-- [hunter] Calculating Config-SHA1
-- [hunter] HUNTER_ROOT: /Users/sgerbino/Projects/hunter
-- [hunter] [ Hunter-ID: xxxxxxx | Toolchain-ID: 1341f84 | Config-ID: 035647d ]
-- [hunter] RABBITMQ-C_ROOT: /Users/sgerbino/Projects/hunter/_Base/xxxxxxx/1341f84/035647d/Install (ver.: 0.10.0)
CMake Error at CMakeLists.txt:14 (find_package):
  By not providing "Findrabbitmq-c.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "rabbitmq-c", but CMake did not find one.

  Could not find a package configuration file provided by "rabbitmq-c" with
  any of the following names:

    rabbitmq-cConfig.cmake
    rabbitmq-c-config.cmake

  Add the installation prefix of "rabbitmq-c" to CMAKE_PREFIX_PATH or set
  "rabbitmq-c_DIR" to a directory containing one of the above files.  If
  "rabbitmq-c" provides a separate development package or SDK, be sure it has
  been installed.


-- Configuring incomplete, errors occurred!
See also "/Users/sgerbino/Projects/hunter/examples/rabbitmq-c/build/CMakeFiles/CMakeOutput.log".
See also "/Users/sgerbino/Projects/hunter/examples/rabbitmq-c/build/CMakeFiles/CMakeError.log".
```
---

### PR DESCRIPTION
**Pull request by sgerbino**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **[Yes]**

---

* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **[Yes]**

* I've checked this [Git style guide](https://0.readthedocs.io/en/latest/git.html). **[Yes]**
* I've checked this [CMake style guide](https://0.readthedocs.io/en/latest/cmake.html). **[Yes]**
* My change will work with CMake 3.2 (minimum requirement for Hunter). **[Yes]**
* I will try to keep this pull request as small as possible and will try not to mix unrelated features. **[Yes]**

---

Resolves #345.

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -445,7 +445,7 @@ hunter_default_version(pugixml VERSION 1.10-p0)
 hunter_default_version(pybind11 VERSION 2.5.0)
 hunter_default_version(qhull VERSION 7.2.0-p1)
 hunter_default_version(quickjs VERSION 2020-04-12-p0)
-hunter_default_version(rabbitmq-c VERSION 0.10.0)
+hunter_default_version(rabbitmq-c VERSION 0.10.0-773b883-t1)
```
Comment (rbsheth) [inline cmake/configs/default.cmake:448]: `-t1` indicates testing. Since the package works without changes from upstream, just drop the `-t1` here.

**Review by rbsheth — COMMENTED** [review]
@sgerbino Thanks for the PR! Can you check on why all the CI tests are failing? Thanks.

**Comment by sgerbino** [comment]
After investigating the CI failures it seems there are two more issues with the `rabbitmq-c`.

1. The target does not seem to have the appropriate include directory tied to it
2. It uses the systems OpenSSL package rather than Hunters

**Comment by rbsheth** [comment]
@sgerbino Will you have some time to fix these issues?

**Comment by NeroBurner** [comment]
closed due to inacivity

**Pull request closed without merging**